src/forces.f90

```diff
@@ -256,267 +268,626 @@ subroutine force(ux1,uy1,ep1)
 
     call derx (ta1,ux1,di1,sx,ffx,fsx,fwx,xsize(1),xsize(2),xsize(3),0,ubcx)    ! dudx
     call derx (tb1,uy1,di1,sx,ffxp,fsxp,fwxp,xsize(1),xsize(2),xsize(3),1,ubcy) ! dvdx
+    call derx (tc1,uz1,di1,sx,ffxp,fsxp,fwxp,xsize(1),xsize(2),xsize(3),1,ubcz) ! dwdx
     call transpose_x_to_y(ta1,ta2) ! dudx
     call transpose_x_to_y(tb1,tb2) ! dvdx
 
     call transpose_x_to_y(ux1,ux2)
     call transpose_x_to_y(uy1,uy2)
+    call transpose_x_to_y(uz1, uz2)
     call transpose_x_to_y(ppi1,ppi2)
-
-    call dery (tc2,ux2,di2,sy,ffyp,fsyp,fwyp,ppy,ysize(1),ysize(2),ysize(3),1,ubcx) ! dudy
-    call dery (td2,uy2,di2,sy,ffy,fsy,fwy,ppy,ysize(1),ysize(2),ysize(3),0,ubcy)    ! dvdy
-    call transpose_y_to_x(tc2,tc1) ! dudy
-    call transpose_y_to_x(td2,td1) ! dvdy
+    call transpose_y_to_z(ux2, ux3)
+    call transpose_y_to_z(uy2, uy3)
+
+    call dery (td2,ux2,di2,sy,ffyp,fsyp,fwyp,ppy,ysize(1),ysize(2),ysize(3),1,ubcx) ! dudy
+    call dery (te2,uy2,di2,sy,ffy,fsy,fwy,ppy,ysize(1),ysize(2),ysize(3),0,ubcy)    ! dvdy
+    call dery (tf2,uz2,di2,sy,ffyp,fsyp,fwyp,ppy,ysize(1),ysize(2),ysize(3),1,ubcz) ! dwdy
+    call transpose_y_to_x(td2,td1) ! dudy
+    call transpose_y_to_x(te2,te1) ! dvdy
```

Comment:
Store dvdy in td1. It is best to have a PR that keep the 2D force calculation unchanged.